matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/send/UserDraft.kt

```diff
@@ -23,15 +23,18 @@ package org.matrix.android.sdk.api.session.room.send
  * EDIT: draft of an edition of a message
  * REPLY: draft of a reply of another message
  */
-sealed class UserDraft(open val text: String) {
-    data class REGULAR(override val text: String) : UserDraft(text)
-    data class QUOTE(val linkedEventId: String, override val text: String) : UserDraft(text)
-    data class EDIT(val linkedEventId: String, override val text: String) : UserDraft(text)
-    data class REPLY(val linkedEventId: String, override val text: String) : UserDraft(text)
+sealed interface UserDraft {
+    val content: String
+    val messageType: String
+
+    data class REGULAR(override val content: String, override val messageType: String) : UserDraft
```

Comment:
I'm wondering if `Voice` should be a dedicated draft type rather than a `messageType` sub type :thinking:

Reply:
I think the code would be clearer with a VOICE data class (they are allcaps...). But I do not know if we will end up with many other cases. @onurays WDYT?

Reply:
it would also mean we can avoid any database entity changes

Reply:
Good catch. That make sense and yes we are preserving the `draftMode` in `DraftEntity`. So no migration will be needed.